transport/http_transport.go

```diff
@@ -75,15 +75,15 @@ type Transport interface {
 	// called.
 	NewClient(ctx context.Context, peer peer.Info) (peer.Client, error)
 
-	// SpawnServer spawns a server that will handle incoming requests for this transport
+	// ListenAndServe spawns a server that will handle incoming requests for this transport
 	// This is used by daemon and cluster packages to create a cluster of instances using
 	// this specific transport.
-	SpawnServer(ctx context.Context, address string) error
+	ListenAndServe(ctx context.Context, address string) error
 
-	// ShutdownServer shuts down the server started when calling SpawnServer()
-	ShutdownServer(ctx context.Context) error
+	// Shutdown shuts down the server started when calling ListenAndServe()
+	Shutdown(ctx context.Context) error
```

Comment:
@thrawn01 do we have some shutdown timeout(graceful shutdown) concept here or it is not necessary?

Reply:
Are you talking about the `context.Context` ?

Yes graceful shutdown's can sometimes take a while. (if there is a handler that is stuck for instance) In such a case it's always best to provide a way to timeout a shutdown. I've been bit by this before in production where a service got stuck in shutdown and we had to forceably shutdown the service losing all of the data in the internal buffers that should have been sync'd to disk. We started adding `context.Context` to all our shutdown methods after that. It's one of those things that you don't think is an issue, until it is. 

Golang's standard http server `Shutdown()` method takes a `context.Context` for this same reason.